### PR TITLE
fix(irc): don't highlight usernames in URLs

### LINKED ISSUE #37: Some issues with puppet username to discord username translations
I noticed at least 2 issues.

Firstly the regex on https://github.com/OpenTTD/dibridge/blob/main/dibridge/irc.py#L144 fails if puppet username contains `[]`
https://discord.com/channels/142724111502802944/1008473233844097104/1010288501742383104
image vs https://weblogs.openttd.org/openttd/2022/08/19.html#204548-124
```
22:45:48 <glx> my irc client will highlight for both glx and glx[d] (because I used a lazy method) but on discord only the [d] version will trigger
22:46:32 <glx> glx[d]: only this format
``` 
and https://discord.com/channels/142724111502802944/1008473233844097104/1010329543644549200
image
vs https://weblogs.openttd.org/openttd/2022/08/19.html#232854-146
```
01:28:54 <glx> let's try glxd
01:29:04 <glx> and discord notified
```

Secondly, puppet username may be translated when it's unwanted https://discord.com/channels/142724111502802944/1008473233844097104/1011205826792407081
image vs https://weblogs.openttd.org/openttd/2022/08/22.html#093056-72
```
11:30:56 <peter1138> https://fuzzle.org/~petern/ottd/diagstations.png
11:30:58 <peter1138> Such old.
11:31:58 <peter1138> https://fuzzle.org/~petern/ottd/stbr3.png
11:31:59 <peter1138> Hehe
```

### PR DESCRIPTION
**Pull request by TrueBrain**
Fixes #37.

### CODE REVIEW

**Comment by TrueBrain** [comment]
@glx22 : mind giving this a review for me? Tnx :)

File: dibridge/irc.py [hunk 1]
```diff
@@ -156,17 +156,24 @@ async def _send_message(self, discord_id, discord_username, message, is_action=F
...
+                [
+                    re.sub(r"(?<!\w)" + re.escape(puppet._nickname) + r"(?!\w)", f"<@{discord_id}>", part)
+                    if "://" not in part
+                    else part
+                    for part in message.split(" ")
+                ]
+            )
+
+            # On IRC, it is common to do "name: ", but on Discord you don't do that ": " part.
+            if message.startswith(f"<@{discord_id}>: "):
```
Comment (glx22) [inline dibridge/irc.py:175]: I did a quick test in python interpreter, and this replacement seem to be already handled by the previous instruction.
```
>>> import re
>>> nickname = "glx[d]"
>>> message = "glx[d]: test http://glx[d].com/"
>>> message = " ".join([re.sub(r"(?<!\w)" + re.escape(nickname) + r"(?!\w)", f"HL", part) if "://" not in part else part for part in message.split(" ")])
>>> print(message)
HL: test http://glx[d].com/
```

Comment (TrueBrain) [reply dibridge/irc.py:175]: Please read the comment and execute the code. It should show a difference ;)

Comment (glx22) [reply dibridge/irc.py:175]: Oh yes, I carefully read the code after commenting, it's indeed not the same replacement. But that implies IRC user know the discord id of the puppet.

Comment (TrueBrain) [reply dibridge/irc.py:175]: No? The replacement to discord is done first, then it checks if there is a : after the highlight?

Just ran some tests, seems to be doing exactly what it should? I am a bit confused, sorry :p

**Review by glx22 — APPROVED** [review]
Once my brain properly works, it looks fine :)

**Pull request merged**